tests/unittest/_torch/modeling/test_modeling_gemma4.py

```
@@ -4041,5 +4051,47 @@ def test_vision_attn_metadata_max_num_requests_is_not_one(self):
+    ],
+)
+def test_gemma4_defaults_resolve_to_v2(architecture: str, model_cls: type) -> None:
+    """The checkpoint's architectures[0] must map to the class carrying the
+    defaults, and those defaults must survive the production resolution
+    ordering (transceiver first, then KV-cache manager) on the NIXL
+    disaggregated route — the route where a missing transceiver preference
+    silently downgrades the V2 default back to V1."""
+    assert MODEL_CLASS_MAPPING[architecture] is model_cls
+
+    llm_args = TorchLlmArgs(
+        model="/tmp/dummy_model",
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/dummy_model"

(S108)